### PR TITLE
Update release process docs and include release checklist in docs

### LINKED ISSUE #284: Update release process docs and include release checklist in docs
Update the release process docs, include moving the release checklist markdown (used in the 0.1.0 release) into the RTD docs.

### PR DESCRIPTION
**Pull request by jgwest**
Fixes #284

### CODE REVIEW

File: docs/Release-Checklist-Template.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- A release commit exists as the most recent commit of the `release-(version)` branch:
+    - [ ]  A commit with the message of `Release (version)`
+        - Example: `Release v0.1.0`
+        - The version string SHOULD contain the letter `v`, as above. 
+    - The commit should contain changes to:
+        - [ ] `manifests/install.yaml`
+        - [ ] `manifests/install-with-argo-cd.yaml`
+        - [ ] `manifests/base/kustomization.yaml`
+        - [ ] `.github/workflows/ci-build.yaml`
+        - [ ] `docs/Geting-Started.md`
```
Comment (wtam2018) [inline docs/Release-Checklist-Template.md:28]:
```suggestion
        - [ ] `docs/Gettng-Started.md`
```

File: docs/Releasing.md [hunk 2]
```diff
@@ -29,18 +33,34 @@ git log
...
 
-#### 3) Verify that the GitHub action jobs fully passed for the commit
+#### 4) Verify that the GitHub action jobs fully passed for the commit
 
 Visit `https://github.com/argoproj-labs/applicationset/commit/(COMMIT ID)`
 
 Verify there is a green check mark at the top left hand corner, indicating that each of the jobs of the GitHub action successfully passed. If not, you should be able to click the icon to see what failed. 
 
-#### 4) Run the `release.sh` script
+#### 5) Update the `docs/Geting-Started.md` file to point to the release version
```
Comment (wtam2018) [inline docs/Releasing.md:42]:
```suggestion
#### 5) Update the `docs/Getting-Started.md` file to point to the release version
```

File: docs/Releasing.md [hunk 3]
```diff
@@ -29,18 +33,34 @@ git log
...
+
+In `docs/Geting-Started.md`, locate the references to `kubectl apply -f`. These are the commands that users will run to install the ApplicationSet controller.
```
Comment (wtam2018) [inline docs/Releasing.md:44]:
```suggestion
In `docs/Getting-Started.md`, locate the references to `kubectl apply -f`. These are the commands that users will run to install the ApplicationSet controller.
```

Comment (jgwest) [reply docs/Releasing.md]: Done.

File: docs/Releasing.md (hunk 2 above)
Comment (jgwest) [reply docs/Releasing.md]: Done.

File: docs/Release-Checklist-Template.md (hunk 1 above)
Comment (jgwest) [reply docs/Release-Checklist-Template.md]: Done.

**Review by wtam2018 — APPROVED** [review]
LGTM 
Thanks @jgwest !

**Pull request merged**